app/controllers/transactions_controller.rb

```diff
@@ -0,0 +1,34 @@
+class TransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_category, only: %i[index new create]
+
+  def index
+    @transactions = @category.transactions.order(created_at: :desc)
+    @total_amount = @transactions.sum(:amount)
+  end
+
+  def new
+    @transaction = @category.transactions.build
+    @categories = current_user.categories
+  end
+
+  def create
+    @transaction = @category.transactions.build(transaction_params)
+    if @transaction.save
+      redirect_to category_transactions_path(@category), notice: 'Transaction was successfully created.'
+    else
+      @categories = current_user.categories
+      render :new
+    end
+  end
```

Comment:
- [ ] [OPTIONAL] It could be better to implement the action to delete transactions and add a button on your page 😃